thriller-bindings/examples/gemm/gemm_g2r.py

```diff
@@ -45,37 +47,35 @@
     print(gB)
     print(gC)
 
-    MemoryLevel = pythriller.PyMemoryLevel.Register
-    RegGraph = pythriller.PyGraph()
+    RegGraph = Graph()
 
-    NodeA = pythriller.PyNode(rA)
-    NodeB = pythriller.PyNode(rB)
-    NodeAcc = pythriller.PyNode(acc)
+    NodeA = Node.tensor(rA)
+    NodeB = Node.tensor(rB)
+    NodeAcc = Node.tensor(acc)
 
-    GemmNode = pythriller.PyNode.gemm(NodeA, NodeB, NodeAcc)
+    GemmNode = Node.gemm(NodeA, NodeB, NodeAcc)
```

Comment:
I just have some questions due to my curiosity.

1. Does `Tensor(data)` also represent a node in the graph, functioning similarly to a "buffer"?

2. Intuitively, `gemm` is an operation with input and output. How does it differentiate between the input and output?

Reply:
1. A `Tensor` is essentially a Buffer, which I have `renamed`:
```rust
#[pyclass(unsendable, module = "buffer", name = "Tensor")]
pub struct PyBuffer(pub Rc<Buffer>);
```
2. `gemm` takes three parameters: `a` and `b` as inputs, and `c` as the output:
```rust
    #[staticmethod]
    fn gemm(a: PyRef<PyNode>, b: PyRef<PyNode>, c: PyRef<PyNode>) -> Self {
        let access_map = AccessMap::new(0, vec![]);

        let node_a = Rc::clone(&a.0);
        let node_b = Rc::clone(&b.0);
        let node_c = Rc::clone(&c.0);

        let gemm = Gemm::new(vec![node_a, node_b], node_c, Rc::new(access_map));

        let node = ThrillerNode::new(ThrillerNodeInner::Op(Box::new(gemm)));

        PyNode(Rc::new(RefCell::new(node)))
    }
```

Reply:
I see. Thank you.